### DISCUSSION TITLE
I don't get it... a tutorial video would be nice 😉

### DISCUSSION THREAD

**Discussion by andyjeffries** [yippee-fun/phlex#170] [General]
I feel like I'm not sure if this is designed to replace your view layer entirely or to define helpers (components)? If the former, how do I get started? Create an `app/phlex/templates/` folder and put my `index.rb` file in there? What about layouts?

I think I'd like the idea of this, views in pure ruby from both a speed and reuse point of view, but I just don't get how to get started.

I think the project could do with a "Blog application" in 5 minutes, like the old Rails site had, but using Phlex templates and components.  Or even an example repo (I checked under https://github.com/joeldrapper) that implements a blog purely using Phlex best practices.

I'm sure others just get it, but as someone that's been a Rails developer for 15 years, it's not clicking for me...

Thanks for any answers.

**Comment by joeldrapper** [comment]
Hey @andyjeffries thanks for posting this request. I really want to do this.

**Comment by andyjeffries** [comment]
Heard you on Remote Ruby, if you want to arrange a call (you mentioned savvycal, but I couldn’t see it in the readme again yet) to walk me through the basics, I don’t mind then doing you a screencast and tutorial guide?

if you want to do it yourself, that’s fine - but thought I’d offer some help if you can just get me started.

**Reply by joeldrapper** [reply]
Hey, that sounds awesome! You can book some time to pair with me here. https://savvycal.com/joeldrapper/pair

**Reply by andyjeffries** [reply]
Booked, see you next week!

**Comment by DannyBen** [comment]
I love this so much! Very excited to have found this.

I am also missing a "Getting Started" page - I don't need a video, but I could only found documentation on the templating syntax, and not about basic usage, like how to render the templates for example. I am sure I can find it in the code in a few minutes, but I figure a quick Getting Started page on the documentation site could be useful to many.

**Reply by andyjeffries** [reply]
I had a call/walkthrough with Joel today, so I got a walkthrough from the main man himself. Planning on writing a Getting Started guide for the docs that walks you through the basics. 

I’ll then make a screencast of it afterwards, for those that learn that way…

**Reply by DannyBen** [reply]
I am particularly interested in a Sinatra context. I have seen [phlex-sinatra](https://github.com/joeldrapper/phlex-sinatra), and would love to see it referenced / expanded upon in such a guide.

**Reply by benkoshy** [reply]
Appreciate the contribution to Open Source, and everyone's contribution on the forums. I am curious as to how this is different from Github View components - besides the performance aspect already noted. 

Some questions i have:

```rb
# Why Phlex
  #### (a) What is the problem(s) phlex tries to solve?
  ####      - Existing approaches (e.g. presenters, github view components, view helpers ):
  ####           - Where they fall short.
  ####           - how phlex solves those problems?
```

**Comment by johnnynia** [comment]
I've finally got Phlex somewhat up and running, but still feel that I'm missing some gotchas. Some examples and/or tutorials for the most basic use cases would add massive value to this very promising project in my opinion.

Especially for people who don't want to change their whole application's view layer, but maybe just want to add a single phlex component that can be rendered from existing views or helpers.

**Reply by joeldrapper** [reply]
Thanks @johnnynia, I’m working on it! Documentation and guides for Rails are my main priority this week. ❤️

**Reply by andyjeffries** [reply]
Did you get my email with an almost finished tutorial style guide?

**Reply by johnnynia** [reply]
@andyjeffries Would it be possible to submit that as a PR? I'd be happy to follow the tutorial as a noob and provide feedback :)

**Reply by joeldrapper** [reply]
> Did you get my email with an almost finished tutorial style guide?

I did, it's great. It might be a good idea to build a tutorial section into the website somehow.

**Reply by andyjeffries** [reply]
Actually @johnnynia I've just realised, it's a Markdown document and not a Phlex view as the site is built using. The document is:
[Phlex tutorial.md](https://github.com/joeldrapper/phlex/files/9753068/Phlex.tutorial.md)

I'm just missing the last step which I'll either get to this week around work or maybe @joeldrapper will bang out when he converts it or rewrites it or whatever he wants to do with it...

**Reply by evenreven** [reply]
Looking forward to reading it, @andyjeffries! I just wrote down some thoughts myself on my experience so far. Less structured than a tutorial, but maybe you'll find some of it useful @johnnynia! Here: https://github.com/joeldrapper/phlex/discussions/241

**Reply by benkoshy** [reply]
> Actually @johnnynia I've just realised, it's a Markdown document and not a Phlex view as the site is built using. The document is: [Phlex tutorial.md](https://github.com/joeldrapper/phlex/files/9753068/Phlex.tutorial.md)

Here's an [online gist link to the tutorial](https://gist.github.com/benkoshy/872ca1b2ccf48d97e93f81105bf96f1c).

As I understand it, Phlex seems to offer exactly what the Github View component offers - the latter seems to allow both usage of ruby directly, as well as HTML markup to build components. Am curious where the differences are?

**Reply by johnnynia** [reply]
Thank you so much @andyjeffries, @evenreven, @benkoshy & @joeldrapper!  ❤️ My understanding (and also the docs) are way clearer already. Some follow-up questions are popping up already, but I'll have to tinker a bit with the provided insights first :)

**Comment by andyjeffries** [comment]
Will do tomorrow when I’m back on my computer!

**Reply by klippas-rob** [reply]
Hi Andy,
Could you take a look at my question on the Tutorial here please?
https://gist.github.com/benkoshy/872ca1b2ccf48d97e93f81105bf96f1c

Much appreciated.
